sdk/src/Core/Internal/Utils/HostEndPoint.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using Amazon.Runtime.Internal.Util;
+
+namespace Amazon.XRay.Recorder.Core.Internal.Utils
+{
+	/// <summary>
+	/// Represents a endpoint on some network.
+	/// The represented endpoint is identified by a hostname.
+	///
+	/// Internally resolves and caches an ip for the hostname.
+	/// The ip is cached to keep the normal path speedy and non-blocking.
+	/// </summary>
+	public class HostEndPoint
+	{
+		private const int CACHE_TTL = 60;	//Seconds to consider a cached dns response valid
+		private IPEndPoint _ipCache = null;
+		private DateTime? _timestampOfLastIPCacheUpdate = null;
+		private static readonly Logger _logger = Logger.GetLogger(typeof(HostEndPoint));
+		private ReaderWriterLockSlim cacheLock = new ReaderWriterLockSlim();
+
+		/// <summary>
+		/// Create a HostEndPoint.
+		/// </summary>
+		/// <param name="host"></param>
+		/// <param name="port"></param>
+		public HostEndPoint(string host, int port)
+		{
+			Host = host;
+			Port = port;
+		}
+		
+		/// <summary>
+		/// Get the hostname that identifies the endpoint.
+		/// </summary>
+		public string Host { get; }
+		/// <summary>
+		/// Get the port of the endpoint.
+		/// </summary>
+		public int Port { get; }
+
+
+		private bool isIPCacheValid()
```

Comment:
The SDK has all methods starting with uppercase. Lets maintain the consistency for all methods in this PR.

Reply:
Addressed in 7cc3c2f